### PR TITLE
Add the ability for widgets to show an external message

### PR DESCRIPTION
**Pull request by vertein**
A widget might want to display an externally sourced message to a user in some sort of disruptive, very heavy handed type of way to alert users of something very important.

Depending on the accent color:
image
image

----

Contributor License Agreement adherence:

- [x] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
Commitlint feedback from Travis CI :building_construction: 
https://travis-ci.org/uPortal-Project/uportal-app-framework/jobs/304773383#L842-L874

```
vertein/uw-frame/MUMUP-3046-WidgetShowExternalMessage
⧗   input: test(widget): add widget controller test cases
✔   found 0 problems, 0 warnings
⧗   input: test(widget): Adds more unit test cases
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
✖   found 1 problems, 0 warnings
⧗   input: test(widget): Add widget service test cases
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
✖   found 1 problems, 0 warnings
⧗   input: license(header): Add missing license headers
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
✖   type must be one of [build, ci, docs, feat, fix, perf, refactor, revert, style, test] [type-enum]
✖   found 2 problems, 0 warnings
⧗   input: test(widget): Adds widget service unit test
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
✖   found 1 problems, 0 warnings
⧗   input: feat(widget): Add continue button to ext message
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
✖   found 1 problems, 0 warnings
⧗   input: license(header): Add missing license header
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
✖   type must be one of [build, ci, docs, feat, fix, perf, refactor, revert, style, test] [type-enum]
✖   found 2 problems, 0 warnings
⧗   input: feat(widget): add widget ext message directive
✔   found 0 problems, 0 warnings
⧗   input: feat(widget): add external widget message service
✔   found 0 problems, 0 warnings
⧗   input: feat(widget): Add static widget message feed
✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
✖   found 1 problems, 0 warnings
```

File: components/portal/widgets/partials/external-message.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+-->
+<div class="overlay__widget-mode" 
+  ng-hide="message_acknowledged || !widget.externalMessageText.trim()">
+  <div class="overlay-content">
+    <p>{{ widget.externalMessageText | limitTo:225}}</p>
+    <md-button ng-if="widget.externalMessageLearnMoreUrl" 
+      class="md-accent md-raised" id="overlay-learn-more-button" 
+      ng-href={{widget.externalMessageLearnMoreUrl}} aria-label=
+      "Learn more about the important message concerning {{widget.fname}}"
```
Comment (apetro) [inline components/portal/widgets/partials/external-message.html]: Using `fname` in `aria-label` isn't ideal. Maybe the external message source should be able to and encouraged to provide an `aria-label`? Maybe this should dereference the `fname` to the portlet title?

File: components/staticFeeds/sample-widget_message.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "mesg": "Success",
+  "result": [
+    {
+      "longMessage": "Your account will be deactivated soon after some time.",
+      "shortMessage": "Your account will be deactivated soon after some time."
```
Comment (apetro) [inline components/staticFeeds/sample-widget_message.json]: Might be best for the sample to differentiate `longMessage` and `shortMessage` so it more effectively demonstrates how the two message lengths are used differently.

File: components/portal/widgets/partials/external-message.html (hunk 1 above)
Comment (thevoiceofzeke) [reply components/portal/widgets/partials/external-message.html]: Or use `widget.title` which should be available.

File: components/portal/widgets/partials/external-message.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      target="_blank" rel="noopener noreferrer">
+        Learn more
+    </md-button>
+    <md-button class="md-default md-raised" id="overlay-continue-button"
```
Comment (thevoiceofzeke) [inline components/portal/widgets/partials/external-message.html]: FWIW: For notifications and announcements, we are using `md-accent md-raised` for the action button and default for the learn more button (`md-default` doesn't actually do/mean anything and omitting the `class` attribute will result in a flat, white button with black text).

File: components/portal/widgets/partials/widget-card.html [hunk 4]
```diff
@@ -21,15 +21,20 @@
 <md-card class="widget-frame" id="widget-id-{{ widget.nodeId }}" aria-label="{{ widget.title }} widget">
 
   <!-- MAINTENANCE MODE OVERLAY -->
-  <div class="overlay__maintenance-mode" ng-if="widget.lifecycleState === 'MAINTENANCE'">
-    <div class="maintenance-content">
+  <div class="overlay__widget-mode" ng-if="widget.lifecycleState === 'MAINTENANCE'">
```
Comment (thevoiceofzeke) [inline components/portal/widgets/partials/widget-card.html]: Why the change in class names?

File: components/portal/widgets/partials/external-message.html (hunk 3 above)
Comment (vertein) [reply components/portal/widgets/partials/external-message.html]: Nice.  I had some feedback to change these to all flat buttons, so that helps

File: components/portal/widgets/partials/widget-card.html (hunk 4 above)
Comment (vertein) [reply components/portal/widgets/partials/widget-card.html]: If we do end up using this for more than just maintenance mode, for example, using it for a widget message, just trying to make sure that the naming is tied to one purpose.

File: components/portal/widgets/partials/external-message.html (hunk 1 above)
Comment (vertein) [reply components/portal/widgets/partials/external-message.html]: widget.title seems like a good call here.

File: components/staticFeeds/sample-widget_message.json (hunk 2 above)
Comment (vertein) [reply components/staticFeeds/sample-widget_message.json]: Good call. +1
